socorro/lib/datetimeutil.py

```diff
@@ -172,3 +173,14 @@ def datestring_to_weekly_partition(date_str):
     last_monday = d + datetime.timedelta(0 - d.weekday())
 
     return last_monday.strftime('%Y%m%d')
+
+
+def datetime_to_midnight(date):
+    """Return the midnight datetime of a given datetime.
+
+    Essentially all this does is return the same date with hour, minute,
+    second and microsecond set to 0.
+    """
+    return datetime.datetime(
+        date.year, date.month, date.day, tzinfo=date.tzinfo
+    )
```

Comment:
I'm sceptical to "polluting" this lib with a function that is only used in one place. 
Normally you use `mydatetimeinstance.replace` for this kind of thing. 

In your `test_get_with_date_histogram` all you need to do is add a little function that looks like this:

```python
def dt_to_midnight(d):
    return d.replace(hour=0, minute=0, second=0, microsecond=0)
```

My concern is that we're modifying a core lib just for the sake of a test. 
That's one thing. Another thing is that if we (some other developer) some day refactors `test_get_with_date_histogram` so it no longer needs `datetimeutil.datetime_to_midnight` then that person might forget to spot that `datetimeutil.py` can be cleaned up and slimmed down. And before you know it grows and becomes dead weight. 

That's why I always think you should only add things to your `main.css` if there's CSS you need on every (or nearly every) page.